src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -573,12 +573,15 @@ ResultWithMessage WindowEditorObjectSelectionSelectObject(
 
     ObjectType objectType = item->Type;
     uint16_t maxObjects = object_entry_group_counts[EnumValue(objectType)];
-
-    if (maxObjects <= _numSelectedObjectsForType[EnumValue(objectType)])
+    if (maxObjects <= _numSelectedObjectsForType[EnumValue(objectType)] && objectType != ObjectType::Water)
     {
         return ObjectSelectionError(isMasterObject, STR_OBJECT_SELECTION_ERR_TOO_MANY_OF_TYPE_SELECTED);
     }
-
+    if (objectType == ObjectType::Water)
+    {
+        // Replace old palette with newly selected palette immediately.
+        ReplaceSelectedWaterPalette(item);
+    }
```

Comment:
Moving this here does not change the logic.

Reply:
this is not a move, rather a hack to bypass the too many objects of this type selected error

Reply:
Surely that is resolved from the other change you did, not this particular one.